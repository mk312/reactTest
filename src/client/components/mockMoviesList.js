const moviesList = {
  data: [
    {
      id: 1,
      title: 'Kill Bill-1',
      tagline: 'Oscar winning',
      vote_average: 5,
      vote_count: 0,
      release_date: '2000',
      poster_path: 'media/51fY8NCyXpL._SX342_.jpg',
      overview: ' American martial arts film written and directed by Quentin Tarantino. '
      + 'It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, '
      + 'Daryl Hannah, and Vivica A. Fox) and their leader, Bill (David Carradine), after they try to kill her and her '
      + 'unborn child. Her journey takes her to Tokyo, where she battles the yakuza.',
      budget: 0,
      revenue: 0,
      runtime: 154,
      genres: [
        'Drama',
      ],
    },
    {
      id: 2,
      title: 'Kill Bill-2',
      tagline: 'string',
      vote_average: 0,
      vote_count: 0,
      release_date: '2002',
      poster_path: 'media/51fY8NCyXpL._SX342_.jpg',
      overview: 'string',
      budget: 0,
      revenue: 0,
      runtime: 0,
      genres: [
        'Drama',
        'Thriller',
      ],
    },
  ],
};

export default moviesList;
