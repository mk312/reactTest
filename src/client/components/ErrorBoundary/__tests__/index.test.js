import React from 'react';
import  ReactDOM from "react-dom";
import {Simulate} from 'react-dom/test-utils';
import {render, fireEvent, screen} from '@testing-library/react'

import { moviesList } from './../../mockMoviesList';

import ErorBoundary from '../index';

let container = null;
let BrokenElement = function BrokenElement() {
    return <div>{props.error}</div>;
};
let Element = function Element() {
    return <div></div>;
};

describe('ErorBoundary', () => {
    it('should show Error notification for user', () => {
        container = render(<ErorBoundary><BrokenElement/></ErorBoundary>).container;
        expect(container).toMatchSnapshot();
    });
    it('should not show anything', () => {
        container = render(<ErorBoundary><Element/></ErorBoundary>).container;
        expect(container).toMatchSnapshot();
    });
});