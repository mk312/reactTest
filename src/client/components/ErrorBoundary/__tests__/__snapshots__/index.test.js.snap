// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErorBoundary should not show anything 1`] = `
<div>
  <div />
</div>
`;

exports[`ErorBoundary should show Error notification for user 1`] = `
<div>
  <div
    class="error"
  >
    Something goes wrong, reload the page please to proceed.
  </div>
</div>
`;
